examples/hosts/app-integration/schema-upgrade/src/modelLoading/modelLoader.ts

```diff
@@ -3,24 +3,33 @@
  * Licensed under the MIT License.
  */
 
-import {
-    IHostLoader,
-} from "@fluidframework/container-definitions";
+import { IContainer, IHostLoader } from "@fluidframework/container-definitions";
 import { ILoaderProps, Loader } from "@fluidframework/container-loader";
+import { IContainerRuntime } from "@fluidframework/container-runtime-definitions";
 import { IRequest } from "@fluidframework/core-interfaces";
 import { ensureFluidResolvedUrl } from "@fluidframework/driver-utils";
-import { IModelCodeLoader, IModelLoader } from "./interfaces";
+import { requestFluidObject, RequestParser } from "@fluidframework/runtime-utils";
+import { IModelLoader } from "./interfaces";
+
+export type ModelMakerCallback<ModelType> = (runtime: IContainerRuntime, container: IContainer) => Promise<ModelType>;
+export const makeModelRequestHandler = <ModelType>(modelMakerCallback: ModelMakerCallback<ModelType>) => {
+    return async (request: RequestParser, runtime: IContainerRuntime) => {
+        if (request.pathParts.length === 0 && request.headers?.containerRef !== undefined) {
```

Comment:
Could be nice to have a comment explaining this condition

Reply:
Added in latest